### PR TITLE
recipe: Add libnl dependency on Linux

### PR DESCRIPTION
**Pull request by dalcinl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,6 +56,7 @@ outputs:
         #- prrte
         - libhwloc
         - libevent
+        - libnl  # [linux]
```
Comment (leofang) [inline recipe/meta.yaml:59]: Nit: I think all three of them have run_exports set up, so we don't need to add them manually.
```suggestion
```

**Review by leofang — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (dalcinl) [reply recipe/meta.yaml:59]: I do not understand your suggestion. This is the `run` section, not the `run_constrained` one. We definitely need libnl as a runtime dependency.

Comment (dalcinl) [reply recipe/meta.yaml:59]: The actual problem is: I do not understand this stuff in full, and with all the other related and unrelated mess, I get very very frustrated.

Comment (leofang) [reply recipe/meta.yaml:59]: "run_exports" is used to ensure ABI compatibility by conda-build. If a package is run_export'd, it means when used in the host section, conda-build will also add it to the run section, with proper pinning constraint set up. So we don't have to worry about it and let the upstream recipe maintainers (in this case, libnl et al) decide the pinning for us.

Comment (leofang) [reply recipe/meta.yaml:59]: If a package is not run_export'd, what we do here is wrong. We'd need to use pin_compatible to declare the pinning manually ourselves. It cannot be free dependencies without constraints.

**Comment by dalcinl** [comment]
@leofang What about zlib? Should we perhaps use libzlib in the run section to not depend in the whole zlib package? https://github.com/conda-forge/zlib-feedstock/blob/main/recipe/meta.yaml

**Comment by leofang** [comment]
Took a quick look, we need `zlib` in host, because it offers the headers. But we can remove it from run, because `zlib` run_exports libzlib

**Review by leofang — APPROVED** [review]

**Comment by leofang** [comment]
It's just time out, let's merge and retry on main. Thanks, Lisandro!

**Pull request merged**